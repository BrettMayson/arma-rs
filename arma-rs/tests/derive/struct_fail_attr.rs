use arma_rs_proc::{FromArma, IntoArma};

#[derive(IntoArma, FromArma)]
#[arma]
pub struct NoList;

#[derive(IntoArma, FromArma)]
#[arma("literal")]
pub struct Literal;

#[derive(IntoArma, FromArma)]
#[arma(unknown)]
pub struct Unknown;

#[derive(IntoArma, FromArma)]
#[arma(unknown::path)]
pub struct UnknownPath;

#[derive(IntoArma, FromArma)]
#[arma(duplicate, duplicate)]
pub struct Duplicate;

#[derive(IntoArma, FromArma)]
#[arma(duplicate::path)]
#[arma(duplicate::path)]
pub struct DuplicatePath;

pub fn main() {}
