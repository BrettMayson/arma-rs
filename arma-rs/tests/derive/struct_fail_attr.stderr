error: expected #[arma(...)]
 --> tests/derive/struct_fail_attr.rs:4:3
  |
4 | #[arma]
  |   ^^^^

error: unexpected literal in nested attribute
 --> tests/derive/struct_fail_attr.rs:8:8
  |
8 | #[arma("literal")]
  |        ^^^^^^^^^

error: unknown attribute `unknown`
  --> tests/derive/struct_fail_attr.rs:12:8
   |
12 | #[arma(unknown)]
   |        ^^^^^^^

error: unknown attribute `unknown::path`
  --> tests/derive/struct_fail_attr.rs:16:8
   |
16 | #[arma(unknown::path)]
   |        ^^^^^^^^^^^^^

error: duplicate attribute `duplicate`
  --> tests/derive/struct_fail_attr.rs:20:19
   |
20 | #[arma(duplicate, duplicate)]
   |                   ^^^^^^^^^

error: duplicate attribute `duplicate::path`
  --> tests/derive/struct_fail_attr.rs:25:8
   |
25 | #[arma(duplicate::path)]
   |        ^^^^^^^^^^^^^^^
